### ISSUE TITLE
O que fazer para iniciar projetos pessoais para um portfólio?

### ISSUE THREAD

**Issue by dyhalmeida** [training-center/forum#56]
Eu tenho alguma dificuldade para juntar tudo que aprendi "solto" de uma linguagem e fazer projetos. As vezes penso, como vou fazer um projeto que possua, estoque, caixa, financeiro e etc, se eu não conheço como funciona um financeiro, um caixa e etc. Como conseguir aprender essas "regras de negócios?" Qual o caminho das pedras para se juntar tudo que aprendeu e colocar isso em prática através de um projeto?
Detalhe, como conseguir essa experiência sozinho, de construir projetos, já que o objetivo é entrar para uma empresa.

**Comment by mayronceccon** [comment]
@dyhalmeida na minha concepção tente iniciar em projetos/módulos pequenos, um projeto desse porte é algo avançado. Outro ponto é que você está fazendo isto para aprendizado, como você não tem uma base concreta, se baseie no SEU entendimento do que cada módulo deve fazer. Com o pontapé inicial você começará a entender o que cada módulo irá precisar e buscará este conhecimento.

**Comment by luizbills** [comment]
Assim como o mayronceccon, recomendo fortemente começar por algo bem pequeno. Só você nunca esquecer que um sistema grande é composto por vários sistemas menores.

**Comment by shadowmaru** [comment]
Uma dica é você procurar uma necessidade SUA, e fazer um projeto em cima disso, pra ter motivação. 

Por exemplo, eu queria aprender React Native. A EMTU, empresa de ônibus intermunicipais de São Paulo, nunca chegou a atualizar seu app pro iOS 11 (apesar de sempre dizerem que o """departamento de informática""" estava vendo isso). Daí eu fui lá e consegui acesso à API deles, criei um app com `create-react-native-app` e fiz um app que mostra os ônibus em tempo real. Lógico que precisa de um monte de coisa a mais pra ele ficar utilizável, e até encontrei outros apps que já funcionavam no iOS 11. Mas a base está lá, e posso dizer que já fiz um app em React Native.

**Comment by dyhalmeida** [comment]
Ótimas dicas, obrigado!

**Comment by AndersonMarquess** [comment]
Boa noite a todos, essa pergunta do  @dyhalmeida é bem o que eu estou passando atualmente, depois fazer alguns cursos pela internet resolvi juntar tudo o que eu aprendi e tentar criar um projeto sozinho do zero, para assim adquirir mais experiência, dai eu comecei a fazer algo simples, que rodava no "terminal", depois aprendi a como criar interfaces gráficas, depois tive que aprender a como usar a gravação em txt, depois no banco de dados...
**Resumindo**: comece com algo simples que você consiga fazer e aos poucos você vai melhorando ele conforme sentir necessidade, ao melhora-lo você aprenderá algo novo e assim é o ciclo.
Sobre como criar um portfólio, acredito que cada projeto mesmo que simples, se você disponibiliza no github por exemplo, já é um portfólio. 
PS: _Essa é a primeira vez que eu interajo o training-center, espero que não tenha problema em postar aqui pois, eu não fiz cadastro em lugar nenhum._

**Comment by bernardodiasc** [comment]
@dyhalmeida voce perguntou sobre regras de negócios, pra começar, voce pode fazer o próprio portifolio ser um exemplo do portifolio, pois nesse caso voce vai estar sabendo das regras de negócio, já que o negócio em questão é o seu próprio trabalho. É bem na linha que o @shadowmaru e o @AndersonMarquess falaram, vc cria suas necessidades e vai avançando.

Outras formas de se explorar regras de negócios é entrando em projetos com outras pessoas que manjam de outras áreas, podem ser projetos open source ou mesmo projetos de clientes. Um cliente normalmente é alguem que tem domínio em alguma área diferente de TI, e contrata um profissional/empresa de TI pra informatizar as regras desse outro domínio. No decorrer das demandas rola um intercâmbio de ideias e o profissional de TI aprende uma quantidade absurda de coisas do outro domínio, tem que aprender pra poder entregar o serviço!

Uma outra forma é voce trabalhar de fato em outra área por um tempo. Por exemplo, arrume um emprego de caixa, almoxarife, etc. São empregos fáceis de conseguir e voce pode ficar alguns meses aprendendo com a mão na massa, anotando tudo e tentando entender como sistematizar o que não estiver sistematizado ou sacar os sistemas já implementados. Essa dica parece radical, mas é muito legal de fazer, eu mesmo, quando quis aprender mais sobre as regras de negócio do ramo alimentícios consegui trabalhar alguns meses de garçon, pizzaiolo, gerente de restaurante, motorista de delivery, e outras coisas. O aprendizado é violento, c faz seu trampo com um caderninho na mão anotando tudo! hahaha Vale a pena a experiencia.

**Comment by mahenrique94** [comment]
Acho que começar com projetos pequenos, pense em algo para lhe ajudar ou ajudar sua familia, procure por necessidades que podem ser melhoradas com sistemas.

**Comment by samuelsilvadev** [comment]
Concordo com o que a galera falou acima, comece por algo pequeno, simples. Cada vez que fizer alguma feature nova terá aprendido algo a mais.

Atualmente estou focado em frontEnd e estou usando alguns sites para me manter estudando diariamente. 

- [FreeCodeCamp](freecodecamp.com)
- [Javascript30](https://javascript30.com/)
- [Frontend challenges](https://github.com/LFeh/frontend-challenges)
- [30days30sites](https://www.subscribepage.com/30days30sites)

Espero ter contribuido <3

**Comment by dyhalmeida** [comment]
Massa @samuelsilvadev , gostei dos sites, valeu!

**Comment by ghost** [comment]
@dyhalmeida 3 dicas simples:

1. estabeleça metas e deadlines a médio e longo prazo, elas te dão uma perspectiva e te ajudam a medir o teu progresso
1. [divida suas metas em pequenos passos](https://hackergaucho.com.br/posts/10_passos_para_o_back-end/8) em ordem crescente de complexidade. mire pequeno e erre pequeno é o lema dos snippers
1. crie bibliotecas de acordo com a necessidade, [não tente criar um framework inteiro logo de cara](https://hackergaucho.com.br/posts/5_dicas_b%C3%A1sicas_para_o_back-end/9)

**Comment by danilosilvadev** [comment]
https://github.com/danilosilvadev/howtomakeaportfolio

**Comment by iagolaguna** [comment]
Tem o https://frontloops.io/ que é muito bom fiz 4 templates dele já, fiz um post bem legal explicando como ele funciona também caso você tenha interesse https://iagolaguna.com/post/frontloops-o-desafio/
